x-pack/plugins/maps/public/classes/layers/new_vector_layer_wizard/wizard.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Component, Fragment } from 'react';
+import { EuiEmptyPrompt, EuiPanel } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { createNewIndexAndPattern } from './utils/indexing_service';
+import { getFileUpload } from '../../../kibana_services';
+import { RenderWizardArguments } from '../layer_wizard_registry';
+import { VectorLayer } from '../vector_layer';
+import { ESSearchSource } from '../../sources/es_search_source';
+import { ADD_LAYER_STEP_ID } from '../../../connected_components/add_layer_panel/view';
+import { IndexNameFormProps } from '../../../../../file_upload/public';
+
+interface State {
+  indexName: string;
+  indexError: string;
+  indexingTriggered: boolean;
+  indexPatternId: string;
+  indexNameFormComponent: React.ComponentType<IndexNameFormProps> | null;
+}
+
+export class NewVectorLayerEditor extends Component<RenderWizardArguments, State> {
+  private _isMounted: boolean = false;
+
+  state: State = {
+    indexName: '',
+    indexError: '',
+    indexingTriggered: false,
+    indexPatternId: '',
+    indexNameFormComponent: null,
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this._loadIndexNameFormComponent();
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async _loadIndexNameFormComponent() {
+    if (this._isMounted) {
```

Comment:
This is still not right. The `this._isMounted` check has to be done after the await. The point of the check is ensure state is not updated after the component as unmounted. The component could unmount will the async task is completing.

Reply:
Yeah, this all kind of placeholdery tbh. @thomasneirynck [had another suggestion](https://github.com/elastic/kibana/pull/96913#discussion_r622265667) that involved removing this logic and taking a different approach that I didn't quite get to in my last pass. It's queued up for my next pass of changes before I re-request his review